File: src/test/kotlin/it/intre/code/database/reader/config/ColumnTest.java
package it.intre.code.database.reader.config;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ColumnTest {

    @Test
    public void getOutName_returns_alias_if_present() {
        Column column = createColumn("NONNA PAPERA", "CICCIO");
        assertEquals("CICCIO", column.getOutName());
    }

    @Test
    public void getOutName_returns_name_if_null_alias() {
        Column column = createColumn("PAPERONE", null);
        assertEquals("PAPERONE", column.getOutName());
    }

    @Test
    public void getOutName_returns_name_if_empty_alias() {
        Column column = createColumn("PAPERONE", "");
        assertEquals("PAPERONE", column.getOutName());
    }

    @Test
    public void getNameWithAlias_returns_name_when_no_alias() {
        Column column = createColumn("PAPEROGA", null);
        assertEquals("PAPEROGA", column.getNameWithAlias());
    }

    @Test
    public void getNameWithAlias_returns_name_when_alias_is_the_same() {
        Column column = createColumn("PIPPO", "PIPPO");
        assertEquals("PIPPO", column.getNameWithAlias());
    }

    @Test
    public void getNameWithAlias_returns_name_and_alias_when_different() {
        Column column = createColumn("PAPERINO", "PAPERINIK");
        assertEquals("PAPERINO AS PAPERINIK", column.getNameWithAlias());
    }

    static Column createColumn(String name, String alias) {
        Column column = new Column(name);
        column.setAlias(alias);
        return column;
    }
}

File: src/test/kotlin/it/intre/code/database/reader/config/ColumnTest.kt
package it.intre.code.database.reader.config

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test


class ColumnTest {

    @Test
    fun getOutName_returns_alias_if_present() {
        val column = createColumn("NONNA PAPERA", "CICCIO")
        assertEquals("CICCIO", column.outName)
    }

    @Test
    fun getOutName_returns_name_if_null_alias() {
        val column = createColumn("PAPERONE", null)
        assertEquals("PAPERONE", column.outName)
    }

    @Test
    fun getOutName_returns_name_if_empty_alias() {
        val column = createColumn("PAPERONE", "")
        assertEquals("PAPERONE", column.outName)
    }

    @Test
    fun getNameWithAlias_returns_name_when_no_alias() {
        val column = createColumn("PAPEROGA", null)
        assertEquals("PAPEROGA", column.nameWithAlias)
    }

    @Test
    fun getNameWithAlias_returns_name_when_alias_is_the_same() {
        val column = createColumn("PIPPO", "PIPPO")
        assertEquals("PIPPO", column.nameWithAlias)
    }

    @Test
    fun getNameWithAlias_returns_name_and_alias_when_different() {
        val column = createColumn("PAPERINO", "PAPERINIK")
        assertEquals("PAPERINO AS PAPERINIK", column.nameWithAlias)
    }

    @Test
    fun `two columns with same attributes are equal`() {
        val column1 = createColumn("Number 1", "ONE")
        column1.formatSource = "sorcio"
        column1.isAggregate = true
        column1.isDerived = false
        column1.type = "INT"

        val column2 = createColumn("Number 1", "ONE")
        column2.formatSource = "sorcio"
        column2.isAggregate = true
        column2.isDerived = false
        column2.type = "INT"

        assertEquals(column1, column2)
    }

    companion object {
        internal fun createColumn(name: String, alias: String?): Column {
            val column = Column(name)
            column.alias = alias
            return column
        }
    }
}
